Continue design time builds if resolving package assets fails

drewnoakes opened the pull request:
(This is an alternative implementation of #3398.)

Design time builds are stopping if `ResolvePackageAssets` hits an error (dotnet/project-system#4992).

This PR uses the fact that design-time builds set the `ContinueOnError` property to `ErrorAndContinue`.

drewnoakes committed:
Continue design time builds if resolving package assets fails

livarcocc commented:
@drewnoakes are any of these PRs ready? still WIP? which one are you recommending we take?

drewnoakes commented:
@livarcocc I'd like someone more familiar with this area to give some guidance. I thought the approach in this PR might give more consistency, but its tests are failing and I didn't manage to find out why before the end of Friday. #3398 may be the better option.

drewnoakes committed:
Remove defaulting of ContinueOnError property

drewnoakes commented:
With @nguerrera's encouragement, I'll continue this version of the change. It does seem more elegant.

I've verified that this change fixes https://github.com/dotnet/project-system/issues/4992. The DTB completes in full, and the dependencies node behaves correctly when you have an unresolved package.

drewnoakes committed:
Set CheckForDuplicateItemsContinueOnError

We no longer use the CheckForDuplicateItemsContinueOnError property in
the SDK, but the WPF repo has a dependency upon it.

This adds an assignment to keep WPF working. It can be removed if
https://github.com/dotnet/wpf/pull/1235 is merged.

drewnoakes commented:
Turns out the WPF repo uses the `CheckForDuplicateItemsContinueOnError` property. I've filed https://github.com/dotnet/wpf/pull/1235 which does basically the same thing as this PR.

db395a0beb470e30ee28b4edd04e7f048e0bcad5 sets the property to `$(ContinueOnError)` so that WPF continues to work. If they merge the above PR then the temporary assignment can be removed.

nguerrera commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets:
```diff
@@ -274,13 +274,17 @@ Copyright (c) .NET Foundation. All rights reserved.
-      <CheckForDuplicateItemsContinueOnError>false</CheckForDuplicateItemsContinueOnError>
-      <CheckForDuplicateItemsContinueOnError Condition="'$(DesignTimeBuild)' == 'true'">ErrorAndContinue</CheckForDuplicateItemsContinueOnError>
+    <!-- We no longer use the CheckForDuplicateItemsContinueOnError property here, but the WPF repo has a dependency upon it.
+         Remove this assignment if https://github.com/dotnet/wpf/pull/1235 is merged. -->
```
Good catch. I'll merge this now. Let's wait until a few days after WPF merges to remove this, just to avoid a race getting to the core-sdk repo between sdk and wpf and having some bad builds.

nguerrera commented:
> The DTB completes in full, and the dependencies node behaves correctly when you have an unresolved package.

That's great. When I tried a similar change, I could swear I saw the diagnostic items in DTB, but not the Dependencies node.

drewnoakes commented:
> When I tried a similar change, I could swear I saw the diagnostic items in DTB, but not the Dependencies node.

This fixes https://github.com/dotnet/project-system/issues/4992 which is about the warning icon. WIth this in, I can start on https://github.com/dotnet/project-system/issues/3780 which relates to the diagnostics.

Thanks again for your help with this.
